lib/rentlinx/modules/link_client_methods.rb

```diff
@@ -0,0 +1,72 @@
+module Rentlinx
+  module LinkClientMethods
+    def post_links(links)
+      raise Rentlinx::InvalidObject, links.find { |a| !a.valid? } unless links.all?(&:valid?)
+      raise Rentlinx::InvalidObject, links unless links.all? { |p| p.propertyID == links.first.propertyID }
+      property_links = links.select { |l| l.class == Rentlinx::PropertyLink }
+      unit_links = links - property_links
+
+      post_property_links(property_links) unless property_links.empty?
+      post_unit_links(unit_links) unless unit_links.empty?
+    end
+
+    def get_links_for_property_id(id)
+      data = request('GET', "properties/#{id}/links")['data']
+      data.map do |link_data|
+        if link_data['unitID'].nil? || link_data['unitID'] == ''
+          link_data.delete('unitID')
+          PropertyLink.new(symbolize_data(link_data))
+        else
+          UnitLink.new(symbolize_data(link_data))
+        end
+      end
+    rescue Rentlinx::NotFound
+      return []
+    end
+
+    def get_links_for_unit(unit)
+      links = get_links_for_property_id(unit.propertyID)
+      links.select { |a| defined? a.unitID && a.unitID == unit.unitID }
```

Comment:
Does the letter a have any significance?  In this situation, it's easy to tell what the variable is, but I still prefer names that are at least slightly meaningful.